From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Wed, 18 Mar 2015 23:39:40 +0800
Subject: [PATCH] {LLVMLinux} [BUG/WIP] drivers/misc/lkdtm: fix clang
 -Wnull-dereference

/mnt/data/src/linux/drivers/misc/lkdtm.c:379:3: error: indirection of
non-volatile null pointer will be deleted, not trap
[-Werror,-Wnull-dereference]
                *((int *) 0) = 0;
---
 drivers/misc/lkdtm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/lkdtm.c b/drivers/misc/lkdtm.c
index 5f1a36b8fbb0..7bbe2ad47101 100644
--- a/drivers/misc/lkdtm.c
+++ b/drivers/misc/lkdtm.c
@@ -387,7 +387,7 @@ static void lkdtm_do_action(enum ctype which)
 		WARN_ON(1);
 		break;
 	case CT_EXCEPTION:
-		*((int *) 0) = 0;
+		*((volatile int *) 0) = 0;
 		break;
 	case CT_LOOP:
 		for (;;)
-- 
